File: nodeJs/sources/treina1.js
const fs = require('fs');

// Lê amostras
const linhas = fs.readFileSync('entradas1.txt', 'utf-8').trim().split('\n');
const amostra = linhas.map(linha => {
  const [peso, especie] = linha.split(' ');
  return { peso: parseFloat(peso), especie };
});

// Inicializa bias aleatório
let bias = Math.random();

for (let x = 0; x < 4000; x++) { //ajusta o bias centenas de vezes
  const fruta = amostra[Math.floor(Math.random() * 25)]; //uma laranja aleatoria

  if (fruta.peso > bias) bias += bias * 0.01; //ajusta o bias um pouco a mais
  else if (fruta.peso < bias) bias -= bias * 0.01; //ajusta o bias um pouco a menos
}

console.log(`O valor que diferencia (bias): ${bias}`);

File: python/sources/treina1.py
import random

dataset=[]
#busca os valores no arquivo e armazena em um array 'dataset'
with open("entradas1.txt","r") as f:
    for line in f:
        peso,especie=line.split(' ')
        dataset.append({'peso':float(peso),'especie':especie})

#inicializa o bias com um valor aleatorio
bias=random.random() #valor entre 0 e 1

for x in range(4000): #ajusta o bias centenas de vezes
    amostra=dataset[random.randint(0,24)] #uma laranja aleatória

    if amostra['peso']>bias:bias+=bias*0.01 #um pouco a mais
    if amostra['peso']<bias:bias-=bias*0.01 #um pouco a menos

print(f"O valor que diferencia (bias): {bias:.6f}")
